pkg/security/x509/verify.go

```diff
@@ -58,3 +81,63 @@ func Verify(certificates []*x509.Certificate, certificateAuthorities []*x509.Cer
 	}
 	return certificates[0].Verify(opts)
 }
+
+type CRLVerification struct {
+	Enabled bool
+	Ctx     context.Context
+	Verify  VerifyByCRL
+}
+
+func VerifyChain(chain []*x509.Certificate, capool *x509.CertPool, crlVerify CRLVerification) error {
+	if len(chain) == 0 {
+		return errors.New("certificate chain is empty")
+	}
+	certificate := chain[0]
+	intermediateCAPool := x509.NewCertPool()
+	for i := 1; i < len(chain); i++ {
+		intermediateCAPool.AddCert(chain[i])
+	}
+	_, err := certificate.Verify(x509.VerifyOptions{
+		Roots:         capool,
+		Intermediates: intermediateCAPool,
+		CurrentTime:   time.Now(),
+		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
```

Comment:
If server verifies then client must be there. If client verifies server must be there.